peps/pep-0694.rst

```diff
@@ -283,639 +383,568 @@ the following keys:
-The response body has no content.
+``mechanism`` (**required**)
+    The file-upload mechanisms the client intends to use for this file.
+    This mechanism **SHOULD** be chosen from the list of mechanisms advertised in the
+    :ref:`Publishing Session response body <publishing-session-response>`.
+    A client **MAY** send a mechanism that is not advertised in cases where server operators have
+    documented a new or up-coming mechanism that is available for use on a "pre-release" basis.
 
-If this single-request upload fails, the entire file must be resent in another single HTTP request.
-This is the recommended, preferred format for file uploads since fewer requests are required.
+``metadata`` (**optional**)
+    If given, this is a string value containing the file's `core metadata
+    <https://packaging.python.org/en/latest/specifications/core-metadata/>`_.
 
-As an example, if the client was to upload a 100,000 byte file, the headers would look like:
+Servers **MAY** use the data provided in this request to do some sanity checking prior to allowing
+the file to be uploaded.  These checks may include, but are not limited to:
 
-.. code-block:: email
+- checking if the ``filename`` already exists in a published release;
 
-    Content-Length: 100000
-    Content-Type: application/octet-stream
-    Upload-Length: 100000
-    Upload-Complete: ?1
+- checking if the ``size`` would exceed any project or file quota;
 
-Clients can opt to upload the file in multiple chunks.  Because the upload resource URL provided in
-the metadata response will be unique per file, clients **MUST** use the given upload resource URL
-for all chunks.  Clients upload file chunks by sending multiple ``POST`` requests to this URL, with
-one request per chunk.
+- checking if the contents of the ``metadata``, if provided, are valid.
 
-For chunked uploads, the ``Content-Length`` is equal to the size in bytes of the chunk that is
-currently being sent. The client **MUST** include a ``Upload-Offset`` header which indicates the
-byte offset that the content included in this chunk's request starts at, and an ``Upload-Complete``
-header with the value ``?0``.  For the first chunk, the ``Upload-Offset`` header **MUST** be set to
-``0``.  As with single-request uploads, the ``Content-Type`` header is ``application/octet-stream``
-and the body is the raw, unencoded bytes of the chunk.
+If the server determines that upload should proceed, it will return a ``202 Accepted`` response,
+with the response body below.
+The :ref:`status <session-status>` of the session will also include
+the filename in the ``files`` mapping.
+If the server determines the upload cannot proceed,
+it **MUST** return a ``409 Conflict``.
+The server **MAY** allow parallel uploads of files, but is not required to.
+If the server cannot proceed with an upload because
+the ``mechanism`` supplied by the client is not supported
+it **MUST** return a ``422 Unprocessable Entity``.
```

Comment:
I suggest mentioning the `422` first, followed by the `409`:

> If the server cannot proceed with an upload because the ``mechanism`` supplied by the client is not supported, it **MUST** return a ``422 Unprocessable Entity``.  If the server determines the upload cannot proceed for any other reason, it **MUST** return a ``409 Conflict``.  The server **MAY** allow parallel uploads of files, but is not required to.

Reply:
1ff2415c